test/unit_tests/braket/circuits/test_circuit.py

```diff
@@ -104,6 +105,41 @@ def test_equality():
     assert circ_1 != non_circ
 
 
+def test_call():
+    alpha = FreeParameter("alpha")
+    theta = FreeParameter("theta")
+    circ = Circuit().h(0).rx(angle=theta, target=1).ry(angle=alpha, target=0)
+    circ_new = circ(theta=1, alpha=0)
+    expected = Circuit().h(0).rx(angle=1, target=1).ry(angle=0, target=0)
+
+    assert circ_new == expected
+
+
+def test_call_with_default_parameter_val():
+    alpha = FreeParameter("alpha")
+    beta = FreeParameter("beta")
+    theta = FreeParameter("theta")
+    gamma = FreeParameter("gamma")
+    circ = (
+        Circuit()
+        .h(0)
+        .rx(angle=theta, target=1)
+        .ry(angle=alpha, target=0)
+        .ry(angle=beta, target=2)
+        .rx(angle=gamma, target=1)
+    )
+    new_circ = circ(np.pi, theta=1, alpha=0)
+    expected = (
+        Circuit()
+        .h(0)
+        .rx(angle=1, target=1)
+        .ry(angle=0, target=0)
+        .ry(angle=np.pi, target=2)
+        .rx(angle=np.pi, target=1)
+    )
+    assert new_circ == expected
+
```

Comment:
what can we assert about new_circ.parameters ?

Reply:
Added in the assertion that this is empty.